src/Uno.UWP/Globalization/NumberFormatting/NumeralSystemTranslator.cs

```diff
@@ -0,0 +1,169 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Uno.Globalization.NumberFormatting;
+
+namespace Windows.Globalization.NumberFormatting
+{
+	public partial class NumeralSystemTranslator
+	{
+		private string _numeralSystem;
+
+		public NumeralSystemTranslator() : this(new string[] { "en-US" })
+		{
+
+		}
+
+		public NumeralSystemTranslator(IEnumerable<string> languages)
+		{
+			ValidateLanguages(languages);
+
+			Languages = languages.ToList();
+			ResolvedLanguage = NumeralSystemTranslatorHelper.GetResolvedLanguage(Languages[0]);
+			NumeralSystem = NumeralSystemTranslatorHelper.GetNumeralSystem(Languages[0]);
+		}
+
+		public string NumeralSystem
+		{
+			get => _numeralSystem;
+			set
+			{
+				if (value is null)
+				{
+					throw new ArgumentNullException("Value cannot be null.");
+				}
+
+				_numeralSystem = NumeralSystemTranslatorHelper.ToPascalCase(value);
+			}
+		}
+
+		public IReadOnlyList<string> Languages { get; }
+
+		public string ResolvedLanguage { get; }
+
+		private void ValidateLanguages(IEnumerable<string> languages)
+		{
+			if (languages is null)
+			{
+				throw new NullReferenceException("Invalid pointer");
```

Comment:
Should this be `NullArgumentException`?

Reply:
I added a test for this at lin 39 in Given_NameralSystemTranslator.cs